lib/msal-core/src/UserAgentApplication.ts

```diff
@@ -2617,5 +2624,34 @@ export class UserAgentApplication {
     return eQParams;
   }
 
+  private setTelemetryManager(config: TelemetryOptions) {
+    // TODO, decide whether or not I need to set a default manager
+    const { applicationName, applicationVersion, telemetryEmitter } = config;
+    const missingVals = []
+    if (!applicationName) {
+      missingVals.push("applicationName (string)");
+    }
+    if (!applicationVersion) {
+      missingVals.push("applicationVersion (string)");
+    }
+    if (telemetryEmitter) {
+      missingVals.push("telemetryEmitter (function)");
+    }
+    if (missingVals.length > 0) {
+      throw ClientConfigurationError.createTelemetryConfigError(missingVals);
+    }
```

Comment:
Another approach for this validation code would be to simplified the code here, and then pass `config` to `createTelemetryConfigError`. One thing I don't like about this code if you have error message business logic (i.e. the strings themselves) mixed into this function, when they could be abstracted into the function that creates the error.

```js
if (!applicationName || !applicationVersion || ! telemetryEmitter) {
 throw ClientConfigurationError.createTelemetryConfigError(config);
}
```

```js
static createTelemetryConfigError(config: TelemetryOptions): ClientConfigurationError {
        const { code, desc } = ClientConfigurationErrorMessage.telemetryConfigError;
        const requiredKeys = {
            applicationName: "string",
            applicationVersion: "string",
            telemetryEmitter: "function"
        }

       const missingKeys = Object.keys(requiredKeys).reduce((keys, key) => config[key] ? keys : keys.concat([ `${key} (${requiredKeys[key]})` ], []);

        const errorMessage = 
        return new ClientConfigurationError(code, `${desc} mising values: ${missingKeys.join(",")}`);
    }
```

Reply:
sold